bip-musig2.mediawiki

```diff
@@ -146,14 +156,16 @@ Note that partial signatures are ''not'' signatures.
 An adversary can forge a partial signature, i.e., create a partial signature without knowing the secret key for the claimed public key<ref>Assume an adversary wants to forge a partial signature for public key ''P''. It joins the signing session pretending to be two different signers, one with public key ''P'' and one with another public key. The adversary can then set the second signer's nonce such that it will be able to produce a partial signature for ''P'', but not for the other claimed signer. An explanation of the individual steps required to create a partial signature forgery can be found in [https://gist.github.com/AdamISZ/ca974ed67889cedc738c4a1f65ff620b this writeup].</ref>.
 However, if ''PartialSigVerify'' succeeds for all partial signatures then ''PartialSigAgg'' will return a valid Schnorr signature.
 
+> On this topic, should we consider making an MuSig aggregated signature over a set of keys a "good enough" proof of knowledge of each indiividual key i.e. a "proof of possesion". We need to do this in DLCs at the moment to prevent some "orcale" on "oracle" key cancellation attacks where you need a PoP for multiple points. I think the security result directly implies this but want to check if we all agree.
+
```

Comment:
I would not want to make a claim about this until we have a proper answer to this question. Here's an attempt at a partial answer. Model the MuSig protocol as an interaction between prover and verifier as follows:
```
Prover sends vector X = (X1, ..., Xn) of "public keys"
Verifier draws random vector b = (a1, ..., an) of keyagg coefficients and replies with b
Prover draws k and sends R = k*G
Verifier sends c
Prover send s
Verifier checks that s*G = R + c*(a1*X1 + ... + an*Xn)
```

With `m = 2*n` transcripts
```
(X, b1, k1, c1, s1)
(X, b1, k1, c2, s2)
(X, b2, k2, c3, s3)
(X, b2, k2, c4, s4)
...
(X, bn, kn, cm, sm)
```
that pass verification, we can build a system of `2*n = m` linear equations in the unknowns `k1, ..., kn` and unknown vector `x` which corresponds to the DLs of X's elements:
```
s1 = k1 + c1*<b1,x>
s2 = k1 + c2*<b1,x>
...
sm = kn + cm*<bn,x>
```

The system of linear equations should be solvable (with overwhelming probability). This is nothing really new compared to the MuSig2 unforgeability proof in the ROM (with "double forking") except that the first fork has `n` branches.

Reply:
I think that makes sense. You seem to be putting together an actual extractability argument. It looks like it could work. My thinking that the "unforgeability" proof would be sufficient to imply it's practical as a proof of possesson i.e. suppose that we give as input a key X* to the adversary and it must create a MuSig PoP signature under an aggregate key with X* as an input key to the aggregation. Clearly this is the same as unforgeability and so implies that they must not be able to create the PoP if MuSig is unforgeable (leaving to the side whether we can construct a proper extractor for the formal definition "proof of knowledge").

Reply:
Yes I think what I said doesn't make sense. I think there is weaker notion where you provide `X*` to an algebraic adversary and they can come up with all the keys themselves **but at least one of the keys must have a non-zero `X*` component** when they present their key list and valid signature under the aggregate key.  Unforgeability in this modified game would imply stronger security properties for MuSig as a PoP (in AGM) because it would mean that any keys that are MuSig aggregated and create a signature must not be an (algebraic) function of any honest party key (unless they had the participation of the honest party in creating the signature). Might need to think on this some more. Not sure it makes sense.